Linking with Cray `ftn` also fails with errors of the form
```
openfms/src/modules/TrajectoryIOModule.f90:115: undefined reference to `fseek_'
```
which appear to be related use of C-like `fseek()` being a GNU extension.

--- a/Makefile
+++ b/Makefile
@@ -59,6 +59,7 @@ SUFFIXES=.f .c .F
 ####################
 
 FMSLIB = src/libfms.a
+MAIN = src/openfms.o
 
 $(FMSLIB): makefmslib
 
@@ -77,7 +78,7 @@ bin:
 bin/$(PROGBASE).zero: $(FMSLIB) $(MAKEFILE) bin
 		@rm -f bin/$(PROGBASE)
 		@echo;echo "Linking $(PROGRAM) ..."
-		$(LD) $(FMSLIB) -I$(MODULEDIR) -o $(TARGET) $(LIBS) $(LDFLAGS)
+		$(LD) $(MAIN) $(FMSLIB) -I$(MODULEDIR) -o $(TARGET) $(LIBS) $(LDFLAGS)
 		@ln -s $(PROGRAM) bin/$(PROGBASE)
 		@echo "done"
 
@@ -85,14 +86,14 @@ bin/$(PROGBASE).zero: $(FMSLIB) $(MAKEFILE) bin
 bin/$(PROGBASE).tc:  $(FMSLIB) $(MAKEFILE) bin
 		@rm -f bin/$(PROGBASE)
 		@echo;echo "Linking $(PROGRAM) ..."
-		$(LD) $(FFLAGS) $(FMSLIB) -o $(TARGET) $(LIBS) $(LDFLAGS)
+		$(LD) $(FFLAGS) $(MAIN) $(FMSLIB) -o $(TARGET) $(LIBS) $(LDFLAGS)
 		@ln -s $(PROGRAM) bin/$(PROGBASE)
 		@echo "done"
 
 bin/$(PROGBASE).quantics: $(FMSLIB) $(MAKEFILE) bin
 		@rm -f bin/$(PROGBASE)
 		@echo;echo "Linking $(PROGRAM) ..."
-		$(LD) $(FMSLIB) -I$(MODULEDIR) -o $(TARGET) $(LIBS) $(LDFLAGS) -J$(QUANTICS_OBJ)/include -L$(QUANTICS_DIR)/bin/dyn_libs -lsrf -lusrf -lsqlite3 -Wl,-rpath=$(QUANTICS_DIR)/bin/dyn_libs
+		$(LD) $(MAIN) $(FMSLIB) -I$(MODULEDIR) -o $(TARGET) $(LIBS) $(LDFLAGS) -J$(QUANTICS_OBJ)/include -L$(QUANTICS_DIR)/bin/dyn_libs -lsrf -lusrf -lsqlite3 -Wl,-rpath=$(QUANTICS_DIR)/bin/dyn_libs
 		@ln -s $(PROGRAM) bin/$(PROGBASE)
 		@echo "done"
 

--- a/src/Makefile
+++ b/src/Makefile
@@ -30,7 +30,7 @@ SUFFIXES=.f .F .f90 .F90
 
 all:		modules $(TARGET)
 
-SRCS		= $(filter-out print_version.F90,$(foreach suffix,$(SUFFIXES),$(wildcard *$(suffix))))
+SRCS		= $(filter-out print_version.F90 openfms.F90,$(foreach suffix,$(SUFFIXES),$(wildcard *$(suffix))))
 MOD_SRCS	= $(foreach suffix,$(SUFFIXES),$(wildcard $(MODULEDIR)/*$(suffix)))
 
 INTFILES        = $(sort $(wildcard *.int))
@@ -78,9 +78,10 @@ DIRTY:=$(shell test -n "$(shell git status --porcelain)" && echo -n '+')
 GIT_VERSION=$(COMMIT)${DIRTY}
 
 # Always recompile print_version.F90 to keep the git commit up-to-date
-print_version.o: print_version.F90 $(OBJS) $(MOD_OBJS)
+print_version.o: print_version.F90 openfms.o $(OBJS) $(MOD_OBJS)
 	$(FC) -c $(FFLAGS) -I $(MODULEDIR) -DGIT_VER="'$(GIT_VERSION)'" $<
 
+openfms.o: $(MOD_OBJS)
 
 %.o : %.F
 	@echo "Compiling $<"

--- a/unit_tests/main.F90
+++ b/unit_tests/main.F90
@@ -7,10 +7,6 @@ program tester
    use test_trajectory, only: collect_trajectory_suite
    use test_bundle, only: collect_bundle_suite
    use GlobalModule, only: set_error_handler
-#ifndef __GNUC__
-! Needed for isatty intrinsic
-   use ifport, only: isatty
-#endif
    implicit none
    integer :: num_failed_tests, is
    type(testsuite_type), allocatable :: testsuites(:)
@@ -20,11 +16,17 @@ program tester
    num_failed_tests = 0
    ! Decide whether the output should be colourful:
    ! 1. Respect the FORCE_COLOR envvar per https://force-color.org/
-   ! 2. Don't color things if the output is not a TTY (e.g. redirected to a file)
+   ! 2. Color things only if the output is a TTY (terminal)
+   ! The latter is currently only checked using the GNU-instrinsic isatty
    call get_environment_variable("FORCE_COLOR", force_color)
-   if (trim(force_color) /= '' .or. (isatty(output_unit) .and. isatty(error_unit))) then
+   if (trim(force_color) /= '') then
       call init_color_output(.true.)
    end if
+#ifdef __GNUC__
+   if (isatty(output_unit) .and. isatty(error_unit)) then
+      call init_color_output(.true.)
+   end if
+#endif
 
    testsuites = [ &
                 new_testsuite("ParticleModule", collect_particle_suite), &

